static/js/components/dashboard/CourseAction.js

```diff
@@ -67,39 +69,39 @@ export default class CourseAction extends React.Component {
     return hasFinancialAid && FA_PENDING_STATUSES.includes(financialAid.application_status);
   }
 
-  renderEnrollButton(run: CourseRun): React$Element<*> {
-    const { openFinancialAidCalculator } = this.props;
-    let text = '';
-    let needsPriceCalculation = this.needsPriceCalculation();
-    let buttonProps = {};
+  handleEnrollButtonClick(run: CourseRun): void {
+    const {
+      coupon,
+      setEnrollSelectedCourseRun,
+      setEnrollCourseDialogVisibility,
+    } = this.props;
+
+    setEnrollSelectedCourseRun(run);
 
-    if (needsPriceCalculation) {
-      text = 'Calculate Cost';
+    if (coupon && isFreeCoupon(coupon)) {
+      this.context.router.push(`/order_summary/?course_key=${encodeURIComponent(run.course_id)}`);
     } else {
-      text = `Pay Now ${this.getCoursePrice()}`;
+      setEnrollCourseDialogVisibility(true);
     }
+  }
+
+  renderEnrollButton(run: CourseRun): React$Element<*> {
+    let buttonProps = {};
 
     if (this.hasPendingFinancialAid()) {
       buttonProps.disabled = true;
-    } else {
-      if (needsPriceCalculation) {
-        buttonProps.onClick = openFinancialAidCalculator;
-      } else {
-        buttonProps.onClick = () => {
-          this.context.router.push(`/order_summary/?course_key=${encodeURIComponent(run.course_id)}`);
-        };
-      }
     }
 
     return (
       <SpinnerButton
-        className="dashboard-button pay-button"
+        className="dashboard-button enroll-button"
         key="1"
         component={Button}
         spinning={run.status === STATUS_PENDING_ENROLLMENT}
+        onClick={() => this.handleEnrollButtonClick(run)}
         {...buttonProps}
       >
-        {text}
+        Enroll Now
```

Comment:
Thanks for catching that! It should be fixed now. When viewing a course that you are already auditing, the button says "Pay Now", and clicking it redirects the user to the order summary page (rather than popping the dialog).